### DISCUSSION TITLE
glob pattern in exceptRoutes field of config/shield.ts

### DISCUSSION THREAD

**Discussion by iammrsea** [adonisjs/core#1399] [Help] [answered]
Is there any way I can add a glob pattern in exceptRoutes in config/shield.ts?
I tried something like this:
```javascript
//config/shield.ts
exceptRoutes: ["/api/v1/**"]
```
but it didn't work. It only worked when I write the route in full. I have quite a few routes to except because my app is both web and api server for a mobile app, and it's a pain to register all the routes individually.

**Comment by McSneaky** [comment]
Don't think so, since it uses [String.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) internally with route pattern

To mach all API routes you can just make it `'/api/v1/'` downside is tho that it also matches `'/foo/bar/api/v1/'`
For more dynamic you can try to use smth like `/api/v1/:id/foo/`, but not sure if that would match `'/api/v1/10/foo/'` or not (I'm not sure what's [route pattern used internally](https://github.com/adonisjs/shield/blob/a93a75804a9f250769cd15fe6874e3591a773ced/src/csrf.ts#L72)

**Reply by iammrsea** [reply]
Followed the link you posted above [route pattern used internally](https://github.com/adonisjs/shield/blob/a93a75804a9f250769cd15fe6874e3591a773ced/src/csrf.ts#L72) and it turned out that it actually uses [Array.prototype.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) and not [String.prototype.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes), consequently, the solution does not work because Array.prototype.includes checks for exact match.
Thanks for the effort though

**Comment by thetutlage** [comment, marked as answer]
Yes, currently shield allows defining `route patterns` and not `glob patterns`. I believe, I can add support for a function that must return a boolean.

```ts
exceptRoutes: (ctx) => {
   // perform your check
   return true // to ignore
   return false // to not ignore
}
```

**Reply by iammrsea** [reply]
That would be great. For the time being, I will make do with typing out all the routes individually.
